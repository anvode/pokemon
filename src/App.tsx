import React from 'react';
import { Route, Switch } from 'react-router-dom';

import PokemonList from './components/PokemonList';
import PokemonDetail from './components/PokemonDetail';
import NotFound from './components/NotFound/NotFound';
import Title from './components/Title/Title';

import './App.scss';

function App() {
    
    return (
        <>
            <Title></Title>
            <div className="app">
                <div className="container-xl pb-4">
                    <Switch>
                        <Route exact path='/' render={() => <PokemonList></PokemonList>} />
                        <Route path='/pokemon/:id' render={() => <PokemonDetail></PokemonDetail>}/>
                        <Route path="*" component={NotFound}/>
                    </Switch>
                </div>
            </div>
        </>
    );
}

export default App;
