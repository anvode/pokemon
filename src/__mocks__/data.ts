export const listResponse = { 
    count: 1,
    next: null,
    previous: null,
    results: [{name: 'bulbasaur', url: 'https://pokeapi.co/api/v2/pokemon/1/'}] 
};

export const chainResponse = {
    'baby_trigger_item': null,
    'chain': {
        'evolution_details': [],
        'evolves_to': [
            {
                'evolution_details': [
                    {
                        'gender': null,
                        'held_item': null,
                        'item': null,
                        'known_move': null,
                        'known_move_type': null,
                        'location': null,
                        'min_affection': null,
                        'min_beauty': null,
                        'min_happiness': null,
                        'min_level': 16,
                        'needs_overworld_rain': false,
                        'party_species': null,
                        'party_type': null,
                        'relative_physical_stats': null,
                        'time_of_day': '',
                        'trade_species': null,
                        'trigger': {
                            'name': 'level-up',
                            'url': 'https://pokeapi.co/api/v2/evolution-trigger/1/'
                        },
                        'turn_upside_down': false
                    }
                ],
                'evolves_to': [
                    {
                        'evolution_details': [
                            {
                                'gender': null,
                                'held_item': null,
                                'item': null,
                                'known_move': null,
                                'known_move_type': null,
                                'location': null,
                                'min_affection': null,
                                'min_beauty': null,
                                'min_happiness': null,
                                'min_level': 36,
                                'needs_overworld_rain': false,
                                'party_species': null,
                                'party_type': null,
                                'relative_physical_stats': null,
                                'time_of_day': '',
                                'trade_species': null,
                                'trigger': {
                                    'name': 'level-up',
                                    'url': 'https://pokeapi.co/api/v2/evolution-trigger/1/'
                                },
                                'turn_upside_down': false
                            }
                        ],
                        'evolves_to': [],
                        'is_baby': false,
                        'species': {
                            'name': 'charizard',
                            'url': 'https://pokeapi.co/api/v2/pokemon-species/6/'
                        }
                    }
                ],
                'is_baby': false,
                'species': {
                    'name': 'charmeleon',
                    'url': 'https://pokeapi.co/api/v2/pokemon-species/5/'
                }
            }
        ],
        'is_baby': false,
        'species': {
            'name': 'charmander',
            'url': 'https://pokeapi.co/api/v2/pokemon-species/4/'
        }
    },
    'id': 2
};

export const chainResponseNoChain = {
    'baby_trigger_item': null,
    'chain': {
        'evolution_details': [],
        'evolves_to': [],
        'is_baby': false,
        'species': {
            'name': 'charmander',
            'url': 'https://pokeapi.co/api/v2/pokemon-species/4/'
        }
    },
    'id': 2
};

export const expectedSetEvolutionChain = [
    {
        'min_level': 1, 
        'species_name': 'charmander', 
        'trigger_name': null, 'url': null
    }, 
    {
        'min_level': 16, 
        'species_name': 'charmeleon', 
        'trigger_name': 
        'level-up', 'url': 'https://pokeapi.co/api/v2/pokemon-species/5/'
    }, 
    {
        'min_level': 36, 
        'species_name': 
        'charizard', 'trigger_name': 
        'level-up', 'url': 'https://pokeapi.co/api/v2/pokemon-species/6/'
    }
];

export const expectedChainResponseNoChain = [
    {
        'min_level': 1,
        'species_name': 'charmander',
        'trigger_name': null,
        'url': null,
    }
];

