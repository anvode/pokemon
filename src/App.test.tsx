import React from 'react';
import { render, cleanup, waitForElement } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import App from './App';
import { listResponse } from './__mocks__/data';
import Store from './redux/store';

describe('App', () => {
    afterEach(cleanup);
    beforeEach(() => {
        fetch.resetMocks();
    });
    it('renders', () => {
        fetch.mockResponseOnce(JSON.stringify(listResponse));

        const { asFragment } = render(<App />, {
            wrapper: function child({ children }) {return <Provider store={Store}><BrowserRouter>{children}</BrowserRouter></Provider>;}
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('inserts text in h1', async () => {
        fetch.mockResponseOnce(JSON.stringify(listResponse));
            
        const { getByTestId } = render(<App />, {
            wrapper: function child({ children }) {return <Provider store={Store}><BrowserRouter>{children}</BrowserRouter></Provider>;}
        });

        expect(getByTestId('loader')).toHaveClass('loader');

        const resolvedH1 = await waitForElement(() => getByTestId('h1'));
        expect(resolvedH1).toHaveTextContent('Pokemon');
        
    });
});