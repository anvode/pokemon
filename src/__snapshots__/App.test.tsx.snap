// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <h1
    class="app__title bg-light p-3 mb-3"
    data-testid="h1"
  >
    Pokemon
  </h1>
  <div
    class="app"
  >
    <div
      class="container-xl pb-4"
    >
      <div
        class="loader__container"
      >
        <div
          class="loader"
          data-testid="loader"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
